sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ConnectionStringProperties.java

```diff
@@ -25,25 +25,31 @@
     private static final String ENDPOINT = "Endpoint";
     private static final String SHARED_ACCESS_KEY_NAME = "SharedAccessKeyName";
     private static final String SHARED_ACCESS_KEY = "SharedAccessKey";
+    private static final String SHARED_ACCESS_SIGNATURE = "SharedAccessSignature";
+    private static final String SAS_VALUE_PREFIX = "sharedaccesssignature ";
     private static final String ENTITY_PATH = "EntityPath";
+    private static final String CS_WITH_ACCESS_KEY = "Endpoint={endpoint};SharedAccessKeyName={sharedAccessKeyName};"
+        + "SharedAccessKey={sharedAccessKey};EntityPath={entityPath}";
+    private static final String CS_WITH_SAS = "Endpoint={endpoint};SharedAccessSignature="
+        + "SharedAccessSignature {sharedAccessSignature};EntityPath={entityPath}";
     private static final String ERROR_MESSAGE_FORMAT = "Could not parse 'connectionString'. Expected format: "
-        + "'Endpoint={endpoint};SharedAccessKeyName={sharedAccessKeyName};"
-        + "SharedAccessKey={sharedAccessKey};EntityPath={entityPath}'. Actual: %s";
+        + CS_WITH_ACCESS_KEY + " or " + CS_WITH_SAS + ". Actual: %s";
     private static final String ERROR_MESSAGE_ENDPOINT_FORMAT = "'Endpoint' must be provided in 'connectionString'."
         + " Actual: %s";
 
     private final URI endpoint;
     private final String entityPath;
     private final String sharedAccessKeyName;
     private final String sharedAccessKey;
+    private final String sharedAccessSignature;
 
     /**
      * Creates a new instance by parsing the {@code connectionString} into its components.
-     *
      * @param connectionString The connection string to the Event Hub instance.
-     * @throws NullPointerException if {@code connectionString} is null.
+     *
```

Comment:
IIRC, our document format is not to align parameters or indent until the parameter.